buildSrc/src/main/groovy/com/uber/okbuck/core/model/base/Scope.groovy

```diff
@@ -108,24 +134,19 @@ class Scope {
             greatest = MultimapBuilder.hashKeys().treeSetValues().build()
         }
 
-        // Download sources if needed
-        if (project.rootProject.okbuck.intellij.sources) {
-            DependencyUtils.downloadSourceJars(project, configurations)
-        }
-
-        LOG.info("Resolving configurations of {} : {}", project, configurations)
-
-        Set<ResolvedArtifactResult> artifacts = configurations.collect {
-            it.incoming.artifacts.artifacts
-        }.flatten() as Set<ResolvedArtifactResult>
+        Set<ResolvedArtifactResult> artifacts = getArtifacts(configurations)
 
         artifacts.each { ResolvedArtifactResult artifact ->
             if (!DependencyUtils.isConsumable(artifact.file)) {
                 return
             }
             ComponentIdentifier identifier = artifact.id.componentIdentifier
```

Comment:
Lets collect these identifiers here in some collection as we are iterating already (if we need to download sources later)